src/azure-cli/azure/cli/command_modules/monitor/operations/general_operations.py

```diff
@@ -0,0 +1,62 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from .._client_factory import cf_monitor
+from ..util import _gen_guid
+from azure.cli.core.commands.transform import _parse_id
+from knack.log import get_logger
+from knack.util import CLIError
+from msrestazure.tools import parse_resource_id
+
+
+logger = get_logger(__name__)
+
+
+def clone_existed_settings(cmd, source_resource=None, target_resource=None):
+    metrics_alert_rules = _clone_monitor_metrics_alerts(cmd, source_resource, target_resource)
+    return metrics_alert_rules
+
+
+def _clone_monitor_metrics_alerts(cmd, source_resource, target_resource):
+    if not _is_resource_type_same_and_sub_same(source_resource, target_resource):
+        raise CLIError('The target resource should be the same type with the source resource '
+                       'and they are in the same subscription')
+    monitor_client = cf_monitor(cmd.cli_ctx, subscription_id=parse_resource_id(source_resource)['subscription'])
+    # we can only clone the alert rules belonging to the source subscription.
+    alert_rules = list(monitor_client.metric_alerts.list_by_subscription())
+    updated_alert_rules = []
+    ErrorResponseException = cmd.get_models('ErrorResponseException', operation_group='metric_alerts')
+    for alert_rule in alert_rules:
+        if source_resource in alert_rule.scopes:
+            if target_resource not in alert_rule.scopes:
+                try:
+                    alert_rule.scopes.append(target_resource)
+                    resource_group_name, name = _parse_id(alert_rule.id).values()
+                    alert_rule = monitor_client.metric_alerts.create_or_update(resource_group_name=resource_group_name,
+                                                                               rule_name=name, parameters=alert_rule)
+                except ErrorResponseException as ex:  # Create new alert rule
+                    if ex.response.status_code == 400:
```

Comment:
>ex.response.status_code == 400 [](start = 23, length = 30)

May I understand in what kind of situation, service return 400?

Reply:
metric alert has different types. Each service support one or some of the types. For instance, metric alert for vm support multiple vms using one metric alert. However, for storage account, it's just a one-one mapping. So if I want to directly add a vm into the scope, it would succeed. If I want to add a storage account into the scope, service would return 400. There might be other situation that server returns 400. So I just have another try to create a new metric alert rule for the target_resource.

Reply:
Can we just create new rules instead of adding resources into scopes? It is easier for user to track the copy result and revert the copy operation.

Reply:
sounds good. The cost of metric alert is not high. let me change it.